qiskit/visualization/circuit/text.py

```diff
@@ -599,67 +599,15 @@ def __init__(
         plotbarriers=True,
         line_length=None,
         vertical_compression="high",
-        layout=None,
         initial_state=True,
         cregbundle=None,
-        global_phase=None,
         encoding=None,
-        qregs=None,
-        cregs=None,
         with_layout=False,
         circuit=None,
```

Comment:
By "make this required", I meant remove the `=None` here and move it up to the top, really - it doesn't have a default value any more.

I wouldn't worry about changing the order of parameters, because for one this is internal anyway, and two you're already changing the order by doing the removals.

Reply:
in 8a558af6a710cfa3db53132fc0134f2ebd197f5d